File: java/selection_sort/src/test/java/com/ahmedsameha1/edx_algorithms_and_data_structures/selection_sort/TestSelectionSort.java
package com.ahmedsameha1.edx_algorithms_and_data_structures.selection_sort;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

public class TestSelectionSort {
    private SelectionSort selectionSort = new SelectionSort();

    @Test
    public void test() {
        assertArrayEquals(new int[] {1, 2, 3, 4}, selectionSort.sort(4, 3, 2, 1));
        assertArrayEquals(new int[] {-1, 1, 2, 3, 4}, selectionSort.sort (4, 3, 2, 1, -1));
        assertArrayEquals(new int[] {-1, 15, 15, 52, 73, 104}, selectionSort.sort(15, 73, 104, 15, 52, -1));
    }
}

File: python/selection_sort/test_selection_sort.py
import unittest
import selection_sort

class TestSelectionSort(unittest.TestCase):
    def test_selection_sort(self):
        self.assertListEqual(selection_sort.sort([4, 3, 2, 1]), [1, 2, 3, 4])
        self.assertListEqual(selection_sort.sort([-47, -168, 111, 4, 3, 10, 2, 1, 18, 1000, 8, 0]), [-168, -47, 0, 1, 2, 3, 4, 8, 10, 18, 111, 1000])

if __name__ == "__main__":
    unittest.main()
